packages/core-plugin-api/api-report.md

```diff
@@ -206,7 +207,7 @@ export function createRouteRef<Params extends {
     id?: string;
     params?: ParamKey[];
     path?: string;
-    icon?: IconComponent;
+    icon?: OldIconComponent;
```

Comment:
This isn't a completely backwards compatible change, since someone might be passing in a component that doesn't have all the props. It's compatible with the `@backstage/core` version though and the field is deprecated, so I think we can call that alright.